mdc_weight_mgmt_xml_import/models/maintenance_equipment.py

```diff
@@ -223,8 +224,28 @@ def _process_xml_file(self, file, processed_path, record):
                 found=equipment.name if equipment else "None"
             ))
         product = self.env['product.product'].search([('default_code', '=', product_code)], limit=1)
-        if not product or not product_code:
-            raise Exception(_("Product with code %(code)s not found", code=product_code))
+        if not product:
+            categ = getattr(record.company_id, 'product_categ_default_id', None)
+            if not categ:
+                categ = self.env.ref("product.product_category_all")
+
+            product_template = self.env['product.template'].create({
```

Comment:
There's a hidden parameter that should be filled (or not): company_id. I've decided to be filled with equipment company, if filled, to be coherent.